File: day05/door_hack.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Digest::MD5 qw(md5_hex);
use 5.24.0;
use feature 'signatures';
no warnings "experimental::signatures";

my $door_id = $ARGV[0];
my $password = '';
my $i = 0;

while (length($password) < 8) {
    my $hash = md5_hex($door_id . $i);
    if (substr($hash, 0, 5) eq '00000') {
        $password .= substr($hash, 5, 1);
        say $password;
    }
    $i++;
}

say $password;

File: day05/door_hack.py
#!/usr/bin/env python
from sys import argv
import hashlib

script, door_id = argv
password = ''
i = 0

while len(password) < 8 :
    k = "%s%i" % (door_id, i)
    m = hashlib.md5(k)
    hash = m.hexdigest()
    if hash[0:5] == "00000" :
        password = password + hash[5]
        print password
    i = i + 1

print password
